qa/tasks/cephadm.py

```diff
@@ -798,6 +798,75 @@ def ceph_iscsi(ctx, config):
 
     yield
 
+@contextlib.contextmanager
+def ceph_ha_rgw(ctx, config):
+    """
+    Deploy haproxy and keepaliveds
+    """
+    cluster_name = config['cluster']
+
+    nodes = []
+    daemons = {}
+    for remote, roles in ctx.cluster.remotes.items():
+        for role in [r for r in roles
+                    if teuthology.is_type('ha-rgw', cluster_name)(r)]:
+            c_, _, id_ = teuthology.split_role(role)
+            log.info('Adding %s on %s' % (role, remote.shortname))
+            nodes.append(remote.shortname + '=' + id_)
+            daemons[role] = (remote, id_)
+    if nodes:
+        # ha rgw can only be deployed using a spec file.
+        # create a sample spec file as a json object
+        ha_rgw_spec = {
+            "service_type": "ha-rgw",
+            "service_id": "haproxy_for_rgw",
+            "placement": str(len(nodes)) + ';' + ';'.join(nodes),
+            "spec": {
+                "virtual_ip_interface": "eth0",
+                "virtual_ip_address": "192.168.20.1/24",
```

Comment:
How do we know what are the name of the interfaces, and the network used? It is fixed always in the theutology env. to the values you have used?